modules/tinymce/src/core/demo/ts/demo/UiSelectionEnabledModeDemo.ts

```diff
@@ -0,0 +1,219 @@
+/* eslint-disable no-console */
+import { Cell, Merger } from '@ephox/katamari';
+import { SugarElement } from '@ephox/sugar';
+
+import { Editor, RawEditorOptions, TinyMCE } from 'tinymce/core/api/PublicApi';
+
+declare let tinymce: TinyMCE;
+
+export default (): void => {
+  const buildModes = (ed: Editor) => {
+    const selectedMode = Cell<string>(ed.mode.get());
+    const setMode = (mode: string) => {
+      ed.mode.set(mode);
+      selectedMode.set(mode);
+    };
+    const makeModeUI = (mode: string) => {
+      const toggleSpec = {
+        text: `${mode} Mode`,
+        onAction: () => {
+          console.log('Current mode: ', ed.mode.get());
+          ed.mode.get() === mode ? ed.mode.set(selectedMode.get()) : setMode(mode);
+          console.log('New mode: ', ed.mode.get());
+        },
+        onSetup: (api: any) => {
+          const toggleActive = () => {
+            api.setActive(ed.mode.get() === mode);
+            api.setEnabled(true);
+          };
+          toggleActive();
+          ed.on('SwitchMode', toggleActive);
+          return () => ed.off('SwitchMode', toggleActive);
+        },
+      };
+      ed.ui.registry.addToggleMenuItem(mode, toggleSpec);
+      ed.ui.registry.addToggleButton(mode, toggleSpec);
+    };
+    ed.mode.register('readonlyUIMode', {
+      activate: () => console.log('Readonly UI: Activated'),
+      deactivate: () => console.log('Readonly UI: Deactivated'),
+      editorReadOnly: true
+    });
+    makeModeUI('readonlyUIMode');
+
+    ed.mode.register('readonlySelectionMode', {
+      activate: () => console.log('Readonly Selection: Activated'),
+      deactivate: () => console.log('Readonly Selection: Deactivated'),
+      editorReadOnly: true
+    });
+    makeModeUI('readonlySelectionMode');
+
```

Comment:
This whole demo might need to be renamed to just readonly mode demo, or something like that.

Reply:
Fixed! Added some demo buttons with different contexts.